src/main/java/org/broadinstitute/hellbender/tools/copynumber/gcnv/IntegerCopyNumberState.java

```diff
@@ -0,0 +1,90 @@
+package org.broadinstitute.hellbender.tools.copynumber.gcnv;
+
+import htsjdk.variant.variantcontext.Allele;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+
+
+/**
+ * This class represents integer copy number states. It is intended to be used as states of a
+ * hidden Markov model.
+ */
+public final class IntegerCopyNumberState  {
+
+    /**
+     * Integer value of the represented copy number state
+     */
+    private final int copyNumber;
+
+    /**
+     * A string representation of this copy number state (used for creating human-readable copy number call lists)
+     */
+    private final String callString;
+
+    /**
+     * An allele representation of this copy number state (used for VCF creation)
+     */
+    private final Allele allele;
+
+    /**
+     * The state with this copy number represents a reference allele
+     */
+    private static final int REFERENCE_COPY_NUMBER_VALUE = 2;
+
+    public IntegerCopyNumberState(final int copyNumber) {
+        this.copyNumber = ParamUtils.isPositiveOrZero(copyNumber, "The integer copy number state" +
+                " must be non-negative");
+        this.callString = toCallString(copyNumber);
+        allele = createAlleleGivenCopyNumber(copyNumber);
+    }
+
+    public Allele toAllele() {
+        return allele;
+    }
+
+    public String getCallString() {
```

Comment:
This method should take `refCopyNumber` and make the call string on the fly accordingly.

Reply:
Done